e2e/framework/eso.go

```diff
@@ -37,7 +37,7 @@ import (
 // with the provided values.
 func (f *Framework) WaitForSecretValue(namespace, name string, expected *v1.Secret) (*v1.Secret, error) {
 	secret := &v1.Secret{}
-	err := wait.PollImmediate(time.Second*10, time.Minute, func() (bool, error) {
+	err := wait.PollUntilContextTimeout(context.Background(), time.Second*1, time.Minute, true, func(ctx context.Context) (bool, error) {
```

Comment:
this definetly needs context awareness for cencallation ability. (i feel like this is a general issue, maybe worth a separate PR)